pkg/serverless/daemon/executioncontext.go

```diff
@@ -0,0 +1,80 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package daemon
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/serverless/executioncontext"
+)
+
+// GetExecutionContext gets the current execution context
+func (d *Daemon) GetExecutionContext() executioncontext.ExecutionContext {
+	d.executionContextMutex.Lock()
+	defer d.executionContextMutex.Unlock()
+	return *d.executionContext
+}
+
+// SetExecutionContextFromInvocation sets the execution context based on an invocation
+func (d *Daemon) SetExecutionContextFromInvocation(arn string, requestID string) {
```

Comment:
since those methods are tied to `ExecutionContext`, maybe we should attach then to a `ExecutionContext` and not the `Daemon`?